### PR TITLE
Upgrade to rlp

### PR DESCRIPTION
**Pull request by zkendra**
- Tests passing
- Need to test with the PCF Tile 
- Need to investigate alternative methods for getting token

### CODE REVIEW

File: ttlcache/ttlcache.go [hunk 1]
```diff
@@ -1,83 +1,122 @@
...
-	cache.mutext.RLock()
-	defer cache.mutext.RUnlock()
+func (c *TTLCache) GetResource(originKey, key string) (resource *results.Resource, found bool) {
+	c.RLock()
+	defer c.RUnlock()
+	return c.getResource(originKey, key)
+}
 
-	if origin, exists := cache.origins[originKey]; exists {
+// private utility func, public methods using it are expected to have mutext RLock minimum
```
Comment (mikehosinski) [inline ttlcache/ttlcache.go]: mutext

Comment (zkendra) [reply ttlcache/ttlcache.go]: ty

File: configuration/configuration_test.go [hunk 2]
```diff
@@ -62,7 +64,7 @@ func TestConfigParsing(t *testing.T) {
 	logger := logger.New(defaultLogDirectory, nozzleLogFile, nozzleLogName, nozzleLogLevel)
 
 	//Create new configuration
-	var config *NozzleConfiguration
```
Comment (seanfisk) [inline configuration/configuration_test.go]: I like the renames.

File: nozzle/nozzle.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+type Nozzle struct {
+	client   *loggregator.RLPGatewayClient
+	config   *configuration.Configuration
+	logger   *gosteno.Logger
+	Messages chan *loggregator_v2.Envelope
+}
+
+func New(config *configuration.Configuration, logger *gosteno.Logger) *Nozzle {
+	l := log.New(NewRLPLogger(logger), "", log.LstdFlags)
```
Comment (seanfisk) [inline nozzle/nozzle.go]: Small suggestion: Use a "fake keyword argument" label to specify the parameter receiving the empty string. [Example from C++ code](https://github.com/BlueMedora/euchre/blob/master/src/gui/views/top_bar.cpp#L32-L33).

File: results/resource.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+func (r *Resource) IsEmpty() bool {
+	r.RLock()
+	defer r.RUnlock()
+	count := 0
+	for _, metrics := range r.ValueMetrics {
+		count += len(metrics)
+	}
+	// count := len(r.valueMetrics)
```
Comment (seanfisk) [inline results/resource.go]: Commented code — do we need this?

File: ttlcache/ttlcache.go [hunk 5]
```diff
@@ -1,83 +1,122 @@
...
+func (c *TTLCache) UpdateResource(e *loggregator_v2.Envelope) {
+	c.Lock()
+	defer c.Unlock()
+
+	k := createEnvelopeKey(e)
+	var r *results.Resource
+	if value, ok := c.getResource(e.Tags["origin"], k); ok {
+		r = value
+	} else {
+		r = results.NewResource(e.Tags["deployment"], e.Tags["job"], e.Tags["index"], e.Tags["ip"])
```
Comment (seanfisk) [inline ttlcache/ttlcache.go]: Is it safe to index into the envelope like this? In other words, if these fields aren't present, will the program crash on an unhandled error?

Comment (zkendra) [reply ttlcache/ttlcache.go]: yes, it is safe, it's not going to work the way we intend, but it won't blow up.

**Pull request merged**